### PR TITLE
Feature 222

### LINKED ISSUE #222: As a guardian I would like to be able to choose different time representations for the timer so that it fits the users preference
**Is your feature request related to a problem? Please describe.**
Citizens want different time representations since some of them might not be able to manage the default option.
**Describe the solution you'd like**
There should be a option to change the timer representations for the citizen in the setting screen as shown below with the "Tidsrepræsentation" button.  
image

When the button is clicked a menu as shown below should appear where it's possible to choose the different time representations.  
image

The timer representation should change for the activity to match the settings. There is quite a bit of work to get the different representations to work. The hourglass needs to act like the circle and sand should gradually fill up in the bottom and become sparse in the top.   

**Additional context**
Some of the other settings shown in the settings screen might not have been implemented. That is not a part of this issue.

### LINKED ISSUE #507: As a citizen i would for the static image on a specific activity depicting the presence of a timer to be the same as the actual timer settings
**Is your feature request related to a problem? Please describe.**
Currently when choosing a new timer representation (numbers, hour glass) they still have the circle icon on the activities. 

**Describe the solution you'd like**
The icon on the activities should be the same as the chosen timer. So if you have chosen a hour glass timer, the icon should also be a hour glass icon.

**Describe alternatives you've considered**
none

**Additional context**
https://cdn.discordapp.com/attachments/687579375776038936/707923401418735707/1.PNG
image

### PR DESCRIPTION
**Pull request by mads-bo**
fix #222

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw612f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [x] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [x] All new functionality is covered by tests
- [x] It is understandable, by reading the tests, what they do
- [x] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [x] Error messages in the code are understandable
- [x] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [x] UI messages, that the user might meet, are actually helpful
- [x] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw613f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [x] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [x] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by holyfailer — DISMISSED** [review]

File: lib/blocs/timer_bloc.dart [hunk 1]
```diff
@@ -43,20 +43,14 @@ class TimerBloc extends BlocBase {
   final BehaviorSubject<bool> _timerInstantiatedStream =
       BehaviorSubject<bool>.seeded(false);
 
+  /// Behavior subject for the progress of the timer in minutes and seconds.
+  final BehaviorSubject<List<int>> _timerProgressNumeric = 
+      BehaviorSubject<List<int>>.seeded(<int>[0, 0, 0]);
```
Comment (KasperAAU) [inline lib/blocs/timer_bloc.dart:49]: Der er her lavet en liste med 3 indexes, men på linje 51-52 der bliver der omtalt at der kun bruges index 0 og 1?
Bliver det tredje index brugt?

**Comment by KasperAAU** [comment]
Det virker til der mangler tests ift. time_representation_screen.dart

File: lib/blocs/timer_bloc.dart (hunk 1 above)
Comment (mads-bo) [reply lib/blocs/timer_bloc.dart:49]: Vi har rettet det til, tidligere havde vi kun minutter og sekunder

**Review by aerlingsson — DISMISSED** [review]

**Review by holyfailer — DISMISSED** [review]

**Review by Lortemanden — CHANGES_REQUESTED** [review]
The icons on the screen where you chose the time representation are too big. They block some of the grey lines as shown in the picture below. This was tested with a screen size off 600x1024, which is our minimum supported screen size. 
image

**Comment by mads-bo** [comment]
Det seneste commit fixes #507

**Review by aerlingsson — DISMISSED** [review]

**Review by Lortemanden — CHANGES_REQUESTED** [review]
The icons look fine in the settings menu. 
There is a problem with the digital clock. When you go to a activity with a digital clock it is set to no time as shown in the picture below.
image

When the timer is then started it shows what the timer is and starts to count down as excepted.
image

If you try to reset the clock it does not change the count to what it was originally and just shows where its stopped. If you then push play it resets the clock and starts to count down.  
image

A last thing is that the representation of the digital clock shown on a activity when viewing a week is chopped so you cant see the 1. 
image

**Review by Lortemanden — DISMISSED** [review]

**Review by aerlingsson — DISMISSED** [review]

**Review by Lortemanden — APPROVED** [review]

**Review by holyfailer — APPROVED** [review]

**Review by aerlingsson — APPROVED** [review]

**Pull request merged**